features/src/autogluon/features/generators/text_special.py

```diff
@@ -138,23 +138,23 @@ def _generate_text_special(self, X: Series, feature: str, symbols: list, X_dict:
         X_dict[fn["char_count"]] = char_count.to_numpy(dtype=np.uint32)
         X_dict[fn["word_count"]] = X_str.str.split().str.len().to_numpy(dtype=np.uint32)
         X_dict[fn["capital_ratio"]] = (
-            X_no_ws.str.count("[A-Z]").divide(X_no_ws_text_len, fill_value=0.0).fillna(0.0).to_numpy(dtype=np.float32)
+            X_no_ws.str.count("[A-Z]").divide(X_no_ws_text_len, fill_value=0.0).fillna(0.0).to_numpy(dtype=np.float64)
```

Comment:
Avoid float64, keep float32, adjust tests accordingly. We don't want to double memory usage. Ditto for rest of cases